### PR TITLE
Add best practices to website

### PR DESCRIPTION
**Pull request by jacobmitchell04**
Based off #58

### CODE REVIEW

File: src/Jobs/BestPracticesUpdater.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    jobFunc() {
+        // eslint-disable-next-line no-async-promise-executor
+        return new Promise(async (resolve, reject) => {
+            try {
+                const data = await axios.get('https://raw.githubusercontent.com/botblock/discord-botlist-best-practices/master/README.md');
+                const bestPractices = await this.db.select().from('kv_cache').where({ key: 'list_best_practices' });
+                if (bestPractices.length) {
+                    await this.db.select().from('kv_cache').where({ key: 'list_best_practices' }).update({ value: data.data });
```
Comment (MattIPv4) [inline src/Jobs/BestPracticesUpdater.js]: Update the datetime here too, so it's always when the kv pair was last updated.

File: src/Routes/Lists.js [hunk 2]
```diff
@@ -278,6 +280,23 @@ class ListsRoute extends BaseRoute {
             }
         });
 
+        this.router.get('/best-practices', async (req, res) => {
+            try {
+                const bestPractices = await this.db.select().from('kv_cache').where({ key: 'list_best_practices' });
+                if (!bestPractices.length) return res.status(400).render('error', {
```
Comment (MattIPv4) [inline src/Routes/Lists.js]:
```suggestion
                if (!bestPractices.length) return res.status(503).render('error', {
```
503 as this is a server-side unavailable error.

File: src/Routes/Lists.js [hunk 3]
```diff
@@ -278,6 +280,23 @@ class ListsRoute extends BaseRoute {
...
+                    title: 'Data not found',
+                    status: 400,
```
Comment (MattIPv4) [inline src/Routes/Lists.js]:
```suggestion
                    status: 503,
```
503 as this is a server-side unavailable error.

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

File: db/migrations/20200114192211_add-kv-cache.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (MattIPv4) [inline db/migrations/20200114192211_add-kv-cache.js]: Extra linebreak here that shouldn't be?

File: src/Dynamic/lists/best_practices.pug [hunk 5]
```diff
@@ -0,0 +1,13 @@
+extends ../layout
+block content
+    section.section
+        .container
+            .hero
+                .hero-body.has-text-centered
+                    h1.title.is-size-1 Best Practices for Discord Bot Lists
+                    h4.title.is-size-4.is-primary Bot lists must meet these minimum requirements to be considered a bot list.
```
Comment (MattIPv4) [inline src/Dynamic/lists/best_practices.pug]: This subtitle is included in the doc itself, feels repetitive?

File: src/Routes/Lists.js [hunk 6]
```diff
@@ -278,6 +280,23 @@ class ListsRoute extends BaseRoute {
...
+            try {
+                const bestPractices = await this.db.select().from('kv_cache').where({ key: 'list_best_practices' });
+                if (!bestPractices.length) return res.status(503).render('error', {
+                    title: 'Data not found',
+                    status: 503,
+                    message: 'Best practices has not yet been loaded.'
+                });
+                res.render('lists/best_practices', {
+                    title: 'Best Practices for Discord Bot Lists',
+                    data: this.renderer.render(bestPractices[0].value)
```
Comment (MattIPv4) [inline src/Routes/Lists.js:293]: Could we some logic to drop the `h1` title from the MD, as it's already rendered in the page itself?

File: src/Jobs/BestPracticesUpdater.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+const BaseJob = require('../Structure/BaseJob');
+const axios = require('axios');
+
+class BestPracticesUpdater extends BaseJob {
```
Comment (MattIPv4) [inline src/Jobs/BestPracticesUpdater.js:4]: This job should run during the application start. We should create a new `Initializers` directory. The migrations check can move to a file there (throw an error if migrations are needed, so it aborts the app boot) & running this job can also be a script in there.

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

File: src/Dynamic/test.pug [hunk 8]
```diff
@@ -4,7 +4,7 @@ block content
         .container
             .hero
                 .hero-body.has-text-centered
-                    h1.title.is-size-1 Test Suite
+                    h1.title.is-size-1 Migrations Suite
```
Comment (MattIPv4) [inline src/Dynamic/test.pug]: Huh?

File: src/Initializers/Migrations.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+class Migrations {
```
Comment (MattIPv4) [inline src/Initializers/Migrations.js:1]: Is there a way to ensure this runs first before anything has a chance to touch the DB?

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

File: src/Initializers/Migrations.js (hunk 9 above)
Comment (jacobmitchell04) [reply src/Initializers/Migrations.js:1]: Sure.

**Review by MattIPv4 — APPROVED** [review]
Lgtm!

**Pull request merged**